storage/memory/admin_storage.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memory
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian"
+	"github.com/google/trillian/storage"
+)
+
+const defaultSequenceIntervalSeconds = 60
+
+// NewAdminStorage returns a storage.AdminStorage implementation backed by
+// memoryTreeStorage.
+func NewAdminStorage(ms storage.LogStorage) storage.AdminStorage {
+	return &memoryAdminStorage{ms.(*memoryLogStorage).memoryTreeStorage}
+}
+
+// memoryAdminStorage implements storage.AdminStorage
+type memoryAdminStorage struct {
+	ms *memoryTreeStorage
+}
+
+func (s *memoryAdminStorage) Snapshot(ctx context.Context) (storage.ReadOnlyAdminTX, error) {
+	return s.Begin(ctx)
+}
+
+func (s *memoryAdminStorage) Begin(ctx context.Context) (storage.AdminTX, error) {
+	return &adminTX{ms: s.ms}, nil
+}
+
+type adminTX struct {
+	ms *memoryTreeStorage
+	// mu guards *direct* reads/writes on closed, which happen only on
```

Comment:
This comment came in with the mysql fork, have you got a suggestion for a better wording?